### PR TITLE
fixes detective holsters super fast

### PR DESCRIPTION
**Pull request by RatFromTheJungle**
you might wanna slap no gbp on it

## About The Pull Request
https://github.com/Skyrat-SS13/Skyrat-tg/issues/21381

## How This Contributes To The Skyrat Roleplay Experience

fixes a buggy-boye, and lets detect holsters hold revolver ammo again (which the sec belt can do anyway)
image
image

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  i don't have to boot it up I know it works
</details>

## Changelog

:cl:
fix: the detective's holster holds ammo again
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/goofsec/code/sec_clothing_overrides.dm
Comment (vinylspiders) [inline modular_skyrat/modules/goofsec/code/sec_clothing_overrides.dm]: This will make them able to hold any type of ammo, including magazines for heavier weapons. I'm not sure if you'd want to limit it to the smaller ones, like shown here with the other detective holster for consistency/so this isn't (potentially) considered a balance change:

https://github.com/Skyrat-SS13/Skyrat-tg/blob/8db30e906d377fa8c335a9c1c8d5c55287778180/code/game/objects/items/storage/holsters.dm#L75-L96

Also you can add 'Fixes' or 'Closed' before the issue # and it will auto-close the linked issue once this gets merged.

Comment (RatFromTheJungle) [reply modular_skyrat/modules/goofsec/code/sec_clothing_overrides.dm]: word

Comment (RatFromTheJungle) [reply modular_skyrat/modules/goofsec/code/sec_clothing_overrides.dm]: i just went back to the previously approved /magazine balance, but added the revolver stuff

**Comment by ghost** [comment]
Don't suppose this fixes the "holds 4 guns" half of the issue? That feels unintended

**Comment by vinylspiders** [comment]
> Don't suppose this fixes the "holds 4 guns" half of the issue? That feels unintended

No, for that you would have to alter the `max_total_storage` and potentially `max_specific_storage` to get your desired effect.

I looked into it and for making it so you can only store 1 (non-bulky) gun in there, it is difficult because of the way weight classes work. Guns that fit in the backpack have a w_class of 3, whereas the some speedloaders and magazines also have a w_class of 3... so you can't really change the `max_total_storage` without permitting more guns to also be stored.

From the item description "A holster able to carry handguns and extra ammo, thanks to an additional hand-sewn pouch." it seems like this could benefit to working like the Crusader's Belt, where you have 1 slot for the gun and then a pouch that can store ammunition. 

image

But honestly that probably isn't needed for this PR. The main issue here was the not being able to store ammo.

**Comment by Stonetear** [comment]
> Don't suppose this fixes the "holds 4 guns" half of the issue? That feels unintended

Check out what I did in [#17278](https://github.com/Skyrat-SS13/Skyrat-tg/pull/17278)
I made a custom storage datum that only allows one gun at a time to be held inside of a storage item.  Just port over the code, and use this storage subtype for whatever items you only want holding one gun at a time.  (It never got merged since I was not willing to make certain design changes, but the code works fine if you want to re-purpose it.)

**Comment by FeenieRU** [comment]
> > Don't suppose this fixes the "holds 4 guns" half of the issue? That feels unintended
> 
> Check out what I did in [#17278](https://github.com/Skyrat-SS13/Skyrat-tg/pull/17278) I made a custom storage datum that only allows one gun at a time to be held inside of a storage item. Just port over the code, and use this storage subtype for whatever items you only want holding one gun at a time. (It never got merged since I was not willing to make certain design changes, but the code works fine if you want to re-purpose it.)

I'm gonna make PR with decreasing slots in holster and rebalance it:
`/obj/item/storage/belt/holster` and `/obj/item/storage/belt/holster/detective` will have 2 and 3 slots, but first can fit only one pistol, second will fit two pistols.

What are you thinking about it and can I use your code?

**Comment by vinylspiders** [comment]
> > > Don't suppose this fixes the "holds 4 guns" half of the issue? That feels unintended
> > 
> > 
> > Check out what I did in [#17278](https://github.com/Skyrat-SS13/Skyrat-tg/pull/17278) I made a custom storage datum that only allows one gun at a time to be held inside of a storage item. Just port over the code, and use this storage subtype for whatever items you only want holding one gun at a time. (It never got merged since I was not willing to make certain design changes, but the code works fine if you want to re-purpose it.)
> 
> I'm gonna make PR with decreasing slots in holster and rebalance it: `/obj/item/storage/belt/holster` and `/obj/item/storage/belt/holster/detective` will have 2 and 3 slots, but first can fit only one pistol, second will fit two pistols.
> 
> What are you thinking about it and can I use your code?

You might as well wait until this fix gets merged first, let them get their GBP. They need it lol.

**Review by jjpark-kb — APPROVED** [review]

**Pull request merged**
